Versión final de la Tarea 3: Derivación por Diferencias Finitas

artcidp opened the pull request:
La tarea ya había sido enviada pero aún no aceptada pues hubo problemas de
conflictos con archivos modificados de las notas de clase. Aquí se
incluye la tarea en una rama nueva, creada a partir de un commit anterior
a dichas modificaciones.

artcidp committed:
Versión final de la Tarea 3: Derivación por Diferencias Finitas

La tarea ya había sido enviada pero aún no aceptada pues hubo problemas de
conflictos con archivos modificados de las notas de clase. Aquí se
incluye la tarea en una rama nueva, creada a partir de un commit anterior
a dichas modificaciones.

UriAceves commented:
Tres comentarios. 

1) Uno está tentado a usar BigFloat porque cree esto **más decimales, más precisión**, hay una diferencia entre precisión y exactitud. En este caso estás ganando precisión y perdiendo exactitud, que no es lo que queremos, de nada nos sirve un valor que tenga un error pequeño, pero prediga un número muy alejado del resultado exacto.

2) Tus gráficas están mal. Sé que uno espera que disminuya el error por siempre, pero eso no es lo que pasa. Pon el resultado sin cucharazos.

3) Es increíblemente falso esto:

> Como se discutió previamente, la derivada simétrica y la derivada compleja resultan en el mismo 
> error, i.e. tienen el mismo grado de exactitud con respecto a la derivada exacta.

artcidp committed:
Correciones: Cambio BigFloat por Float64

Cambié las definiciones de las derivadas de modo que no promovieran los
números a BigFloat. También corregí las gráficas y ahora se puede apreciar
el error numérico.

artcidp committed:
Pequeñas Correcciones

artcidp commented:
Listo. Cambié lo que mencionaste. Avísame si se requieren hacer más cambios.

UriAceves commented:
> Se bede ser cuidadoso

:+1: 

Vientos, todo lo demás te quedó muy chido. Hago el merge.
